xclim/core/units.py

```diff
@@ -582,15 +582,15 @@ def to_agg_units(
     >>> degdays.units
     'K d'
     """
-    if op in ["min", "max", "mean"]:
-        out.attrs["units"] = orig.attrs["units"]
+    if op in ["amin", "min", "amax", "max", "mean", "std"]:
+        out.attrs["units"] = orig.attrs["units"] if "units" in orig.attrs else ""
```

Comment:
```suggestion
        out.attrs["units"] = orig.attrs["units"]
```
This is not supposed to happen! Usually, xclim should have failed before, but we don't want to assume units if None are given (including "dimensionless").